src/domain/dataset-update-flow/src/aggregates/schedule.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::Utc;
+use event_sourcing::*;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Aggregate, Debug)]
+pub struct UpdateSchedule(Aggregate<UpdateScheduleState, (dyn UpdateScheduleEventStore + 'static)>);
```

Comment:
# Generic scheduling service?
I feel like we will have so much more uses for recurrent / cron-based actions:
- poll/push/deriv updates
- compactions?
- replication?
- archival / truncation?
- garbage collection?

Should we make this into a standalone super-generic service, e.g. one that can publish any **command to our domain bus** on a specified schedule?

Note: I am trying hard to resist the urge to over-generalize, but it would also suck to end up with half a dozen cron services for every different kind of upkeep actions.

Reply:
The specific things for now are:
 - weak link with dataset via ID
 - trigger type
 - delay type (wait condition)
 
The last 2 could be generalized, while the 1st thing might be solved via objects composition.
If there are multiple job kinds for the datasets, we can also introduce job type enumeration.